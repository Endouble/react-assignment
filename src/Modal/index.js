import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

const Modal = props => (
    <div className="Modal">
        <div className="Modal__content">
            <button className="Modal__close-container" onClick={props.onCloseClick}>
                <svg className="Modal__close-icon" viewBox="0 0 40 40"><path d="M 10,10 L 30,30 M 30,10 L 10,30" /></svg>
            </button>
            <div className="Modal__body">
                {props.children}
            </div>
        </div>
    </div>
);

Modal.defaultProps = {
    onCloseClick: () => {},
    children: '',
};

Modal.propTypes = {
    onCloseClick: PropTypes.func,
    children: PropTypes.node,
};

export default Modal;
