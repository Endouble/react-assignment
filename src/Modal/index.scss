.Modal {
    background-color: rgba(0,0,0,0.5);
    position: fixed;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;

    &__content {
        position: fixed;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        background-color: #fff;
        box-shadow: 0 0 10px 3px rgba(0, 0, 0, 0.1);
        overflow-y: auto;
        -webkit-overflow-scrolling: touch;

        @media screen and (min-width: 500px) {
            left: 50%;
            top: 50%;
            height: auto;
            transform: translate(-50%, -50%);
            max-width: 500px;
            max-height: calc(100% - 40px);
        }
    }

    &__close-container {
        position: absolute;
        top: 0;
        right: 0;
        padding: 5px;
        line-height: 1;
        background: #f6f6f7;
        border: 0;
        box-shadow: 0;
        cursor: pointer;
    }

    &__close-icon {
        width: 25px;
        height: 25px;
        fill: transparent;
        stroke: black;
        stroke-linecap: round;
        stroke-width: 2;
    }

    &__body {
        padding: 40px;
    }
}
