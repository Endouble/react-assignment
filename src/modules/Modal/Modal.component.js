/* global document */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Button from '../Button';

import { colors, sizes } from '../../styles/vars';

class Modal extends React.Component {
    constructor(props) {
        super(props);
        this.closeKeyBoard = this.closeKeyBoard.bind(this);
    }
    componentDidMount() {
        document.addEventListener('keydown', this.closeKeyBoard, false);
    }

    componentWillUnmount() {
        document.removeEventListener('keydown', this.closeKeyBoard, false);
    }

    closeKeyBoard(event) {
        const { hideModal } = this.props;
        // ESC keyCode
        if (event.keyCode === 27) {
            hideModal();
        }
    }

    render() {
        const { isOpen, children, hideModal } = this.props;
        return (
            <StyledModal isOpen={isOpen}>
                <div className="pokeModal__close">
                    <Button isPlain={false} onClick={hideModal} backColor={colors.red} />
                </div>
                <div className="pokeModal__shell">
                    {children}
                </div>
            </StyledModal>
        );
    }
}

/*
    Modal Styles
*/
const StyledModal = styled.div`
    background: ${colors.darkGray};
    display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 1;

    .pokeModal__close {
        position: absolute;
        top: 10px;
        right: 10px;

        @media (min-width: ${sizes.tablet}) {
            top: 30px;
            right: 30px;
        }
    }

    .pokeModal__shell {
        max-width: 90%;
        margin: 80px auto 0;
        color: white;
        overflow: auto;
        max-height: 100vh;
        z-index: 10;

        @media (min-width: ${sizes.mobile}) {
            max-width: 60%;
            padding-bottom: 100px;
        }

        @media (min-width: ${sizes.tablet}) {
            margin: 30px auto 0;
            max-width: 40%;
            padding-bottom: 100px;
        }
    }

`;

/*
    Modal propTypes
*/
Modal.propTypes = {
    children: PropTypes.node,
    isOpen: PropTypes.bool,
    hideModal: PropTypes.func,
};

Modal.defaultProps = {
    children: null,
    isOpen: false,
    hideModal: () => null,
};

export default Modal;
