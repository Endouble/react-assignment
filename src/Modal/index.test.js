import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './index.js';

it('Modal renders correctly', () => {
    const props = {
        onCloseClick: () => {},
        children: 'Hello!',
    };

    const tree = renderer
        .create(<Modal {...props} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
