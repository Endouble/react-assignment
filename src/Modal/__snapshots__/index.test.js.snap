// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div
  className="Modal"
>
  <div
    className="Modal__content"
  >
    <button
      className="Modal__close-container"
      onClick={[Function]}
    >
      <svg
        className="Modal__close-icon"
        viewBox="0 0 40 40"
      >
        <path
          d="M 10,10 L 30,30 M 30,10 L 10,30"
        />
      </svg>
    </button>
    <div
      className="Modal__body"
    >
      Hello!
    </div>
  </div>
</div>
`;
